server/src/main/java/org/opensearch/extensions/ExtensionsOrchestrator.java

```diff
@@ -347,16 +348,15 @@ TransportResponse handleExtensionRequest(ExtensionRequest extensionRequest) thro
             case REQUEST_EXTENSION_CLUSTER_SETTINGS:
                 ClusterSettingsResponse clusterSettingsResponse = new ClusterSettingsResponse(clusterService);
                 return clusterSettingsResponse;
-            case REQUEST_EXTENSION_ACTION_LISTENER_ON_FAILURE:
-                return handleExtensionActionListenerOnFailureRequest(extensionRequest.getFailureException());
             default:
-                throw new Exception("Handler not present for the provided request");
+                throw new IllegalArgumentException("Handler not present for the provided request");
         }
     }
 
-    private ExtensionBooleanResponse handleExtensionActionListenerOnFailureRequest(String failureException) throws Exception {
+    public ExtensionBooleanResponse handleExtensionActionListenerOnFailureRequest(ExtensionActionListenerOnFailureRequest request)
```

Comment:
Can we create a separate handler class for this handler? Refer to @dbwiddis PR here: https://github.com/opensearch-project/OpenSearch/pull/4519/files -> `SettingsRequestHandler`. 
This will save some work for Frank on https://github.com/opensearch-project/opensearch-sdk-java/issues/116

Reply:
Also see `restActionsRequestHandler.handleRegisterRestActionsRequest(request)))`.